ocaml/xapi/xapi_cluster.ml

```diff
@@ -49,14 +49,21 @@ let create ~__context ~pIF ~cluster_stack ~pool_auto_join ~token_timeout
       let host = Helpers.get_master ~__context in
       let pifrec = Db.PIF.get_record ~__context ~self:pIF in
       assert_pif_prerequisites (pIF, pifrec) ;
-      let ip = ip_of_pif (pIF, pifrec) in
+      let ip_addr = ip_of_pif (pIF, pifrec) in
+      let hostid = Inventory.lookup Inventory._installation_uuid in
+      let hostname = Db.Host.get_hostname ~__context ~self:host in
+      let member =
+        Cluster_interface.(
+          UuidAddr {ip= IPv4 (ipstr_of_address ip_addr); hostname; hostid}
+        )
+      in
       let token_timeout_ms = Int64.of_float (token_timeout *. 1000.0) in
       let token_timeout_coefficient_ms =
         Int64.of_float (token_timeout_coefficient *. 1000.0)
       in
       let init_config =
         {
-          Cluster_interface.local_ip= ip
+          Cluster_interface.local_ip= member
```

Comment:
It's a pity this is called `local_ip` but now has more data. Was it not possible to add additional fields to `init_config` with type string option for compatibility? There is already a `name` that is unused?

Reply:
Actually maybe we can just change the `init_config` altogether since this is only used in a local call, i.e. xapi -> clusterd but not any remote ones